test/extensions/filters/http/jwt_authn/extractor_test.cc

```diff
@@ -172,6 +198,56 @@ TEST_F(ExtractorTest, TestPrefixHeaderMatch) {
   EXPECT_FALSE(headers.get(Http::LowerCaseString("prefix-header")));
 }
 
+// Test extracting token from the custom header: "prefix-header"
+// The value is found after the "CCCDDD", then between the '=' and the ','.
+TEST_F(ExtractorTest, TestPrefixHeaderFlexibleMatch1) {
+  auto headers = TestHeaderMapImpl{{"prefix-header", "preamble CCCDDD=jwt_token,extra=more"}};
+  auto tokens = extractor_->extract(headers);
+  EXPECT_EQ(tokens.size(), 1);
+
+  // Match issuer 7 with map key as: prefix-header + CCCDDD
+  EXPECT_TRUE(tokens[0]->isIssuerSpecified("issuer7"));
+  EXPECT_EQ(tokens[0]->token(), "jwt_token");
+
+  // Test token remove
+  tokens[0]->removeJwt(headers);
+  EXPECT_FALSE(headers.get(Http::LowerCaseString("prefix-header")));
+}
+
+TEST_F(ExtractorTest, TestPrefixHeaderFlexibleMatch2) {
+  auto headers =
+      TestHeaderMapImpl{{"prefix-header", "CCCDDD=\"and0X3Rva2Vu\",comment=\"fish tag\""}};
+  auto tokens = extractor_->extract(headers);
+  EXPECT_EQ(tokens.size(), 1);
+
+  // Match issuer 7 with map key as: prefix-header + AAA
+  EXPECT_TRUE(tokens[0]->isIssuerSpecified("issuer7"));
+  EXPECT_EQ(tokens[0]->token(), "and0X3Rva2Vu");
+
+  // Test token remove
+  tokens[0]->removeJwt(headers);
+  EXPECT_FALSE(headers.get(Http::LowerCaseString("prefix-header")));
+}
+
+TEST_F(ExtractorTest, TestPrefixHeaderFlexibleMatch3) {
+  auto headers = TestHeaderMapImpl{
+      {"prefix-header", "creds={\"authLevel\": \"20\", \"CCCDDD\": \"and0X3Rva2Vu\"}"}};
+  auto tokens = extractor_->extract(headers);
+  EXPECT_EQ(tokens.size(), 2);
+
+  // Match issuer 8 with map key as: prefix-header + "CCCDDD"
+  EXPECT_TRUE(tokens[0]->isIssuerSpecified("issuer8"));
+  EXPECT_EQ(tokens[0]->token(), "and0X3Rva2Vu");
+
+  // Match issuer 7 with map key as: prefix-header + CCCDDD
+  EXPECT_TRUE(tokens[1]->isIssuerSpecified("issuer7"));
+  EXPECT_EQ(tokens[1]->token(), "and0X3Rva2Vu");
+
+  // Test token remove
+  tokens[0]->removeJwt(headers);
```

Comment:
no need to test removeJwt() call in these newly added tests.

Reply:
Good point.

Reply:
Those redundant tests of removeJwt() have been removed in most recent commits.